testsuite/src/it/java/org/graalvm/tests/integration/AppReproducersTest.java

```diff
@@ -467,20 +464,26 @@ public void imageioAWT(TestInfo testInfo, Apps app) throws IOException, Interrup
             // Test static libs in the executable
             final File executable = new File(appDir.getAbsolutePath() + File.separator + "target", "imageio");
             //TODO: This might be too fragile... e.g. order shouldn't matter.
-            final String toFind;
+            final Set<String> expected;
+            //@formatter:off
             if (UsedVersion.getVersion(inContainer).compareTo(Version.create(22, 2, 0)) >= 0) {
-                // libmanagement_ext.a added in 22.2 with https://github.com/oracle/graal/commit/a0e6a3aeb8b63f6c06dc3554c342075534d90796
-                toFind = "libnet.a|libjavajpeg.a|libnio.a|libmanagement_ext.a|liblibchelper.a|libjava.a|liblcms.a|libfontmanager.a|libawt_headless.a|libawt.a|libfdlibm.a|libzip.a|libjvm.a";
-            } else if (UsedVersion.jdkFeature(inContainer) > 11 || UsedVersion.jdkUpdate(inContainer) > 12) {
+             // libmanagement_ext.a added in 22.2 with https://github.com/oracle/graal/commit/a0e6a3aeb8b63f6c06dc3554c342075534d90796
+             // expected = Set.of("libawt.a", "libawt_headless.a", "libfdlibm.a", "libfontmanager.a",                  "libjava.a", "libjavajpeg.a", "libjvm.a", "liblcms.a", "liblibchelper.a", "libmanagement_ext.a", "libnet.a", "libnio.a", "libzip.a");
+             // libmanagement_ext.a removed again: https://github.com/oracle/graal/pull/4383
+                expected = Set.of("libawt.a", "libawt_headless.a", "libfdlibm.a", "libfontmanager.a",                  "libjava.a", "libjavajpeg.a", "libjvm.a", "liblcms.a", "liblibchelper.a",                         "libnet.a", "libnio.a", "libzip.a");
+            } else if (UsedVersion.jdkFeature(inContainer) > 11 || (UsedVersion.jdkFeature(inContainer) == 11 && UsedVersion.jdkUpdate(inContainer) > 12)) {
                 // Harfbuzz removed: https://github.com/graalvm/mandrel/issues/286
-                toFind = "libnet.a|libjavajpeg.a|libnio.a|liblibchelper.a|libjava.a|liblcms.a|libfontmanager.a|libawt_headless.a|libawt.a|libfdlibm.a|libzip.a|libjvm.a";
+                expected = Set.of("libawt.a", "libawt_headless.a", "libfdlibm.a", "libfontmanager.a",                  "libjava.a", "libjavajpeg.a", "libjvm.a", "liblcms.a", "liblibchelper.a",                         "libnet.a", "libnio.a", "libzip.a");
```

Comment:
Nitpick, since the order is not important in the Set why don't you append the additional libraries, to avoid the need for the whitespace (which a formatter might remove)

Reply:
@zakkak I am not sure I understand. I see it as a sort of spreadsheet, immediately seeing which ones are missing.  Do you mean to arrange it in columns as `expected.add()`?

Reply:
@zakkak The whole point of this PR is to make it more understandable, so as one immediately sees what's wrong with the list, so this is actually not a nitpick but an important thing :-D

Reply:
> I see it as a sort of spreadsheet, immediately seeing which ones are missing. 

This only happens in the source code though, right? When printing the Set:
1. The order is not guaranteed
2. the whitespace is not printed

I was thinking about something like:
```
expected = Set.of("libawt.a", "libawt_headless.a", "libfdlibm.a", "libfontmanager.a", "libjava.a", "libjavajpeg.a", "libjvm.a", "liblcms.a", "liblibchelper.a", "libnet.a", "libnio.a", "libzip.a");
```
as the core libraries, and then:
```
expected = Set.of("libawt.a", "libawt_headless.a", "libfdlibm.a", "libfontmanager.a", "libjava.a", "libjavajpeg.a", "libjvm.a", "liblcms.a", "liblibchelper.a", "libnet.a", "libnio.a", "libzip.a", "libharfbuzz.a");
```

or

```
expected = Set.of("libawt.a", "libawt_headless.a", "libfdlibm.a", "libfontmanager.a", "libjava.a", "libjavajpeg.a", "libjvm.a", "liblcms.a", "liblibchelper.a", "libnet.a", "libnio.a", "libzip.a", "libmanagement_ext.a");
```

etc.

>  @zakkak The whole point of this PR is to make it more understandable, so as one immediately sees what's wrong with the list, so this is actually not a nitpick but an important thing :-D

Agreed, wouldn't keeping the "core libraries" as a prefix and then appending the extra libraries for "special" cases have the same result?